pandas/core/dtypes/cast.py

```diff
@@ -353,6 +353,12 @@ def maybe_promote(dtype, fill_value=np.nan):
             fill_value = np.nan
             dtype = np.dtype(np.object_)
 
+    return maybe_promote_scalar(dtype, fill_value)
+
+
+def maybe_promote_scalar(dtype, fill_value=np.nan):
```

Comment:
can you add a doc-string / types

Reply:
will do

Reply:
added docstring, annotations i need to coordinate with others about implementing correct types in pd._typing